src/tv2-common/content/dve.ts

```diff
@@ -483,6 +441,66 @@ export function MakeContentDVE2<
 	}
 }
 
+const setBoxSource = (
+	boxConfig: BoxConfig,
+	boxSources: BoxSources,
+	sourceInfo: { port: number; sourceLayerType: SourceLayerType }
+) => {
+	if (boxConfig) {
+		boxConfig.source = sourceInfo.port
+
+		boxSources.push({
+			// TODO - draw box geometry
+			...boxSource(sourceInfo),
+			...literal<CameraContent | RemoteContent>({
+				studioLabel: '',
+				switcherInput: sourceInfo.port
+			})
+		})
+	}
+}
+
+const setBoxToBlack = (boxConfig: BoxConfig, boxSources: BoxSources) => {
+	setBoxSource(boxConfig, boxSources, {
+		port: AtemSourceIndex.Blk,
+		sourceLayerType: SourceLayerType.UNKNOWN
+	})
+}
+
+function makeBoxMap(
```

Comment:
This function should not be a void function, but instead it should return the `boxMap`.
The prefix `make` works similarly to the prefix `create` and when you call methods with those prefixes you expect that it creates and returns something to you. But this method don't

Instead, it creates 'a side effect'.
Instead of returning the boxMap that it is supposed to 'make', you have to provide it an already created boxMap that it then can 'make' and populate. Aka, it has an effect on something that is outside of the function (this is generally a bad thing, that unfortunately is widespread in Sofie)

What the function does is that it `populates` a boxMap.

Ideally, this function should not take the `boxMap` parameter and instead return the newly created boxMap.
So it would look something like this:
```
function makeBoxMap(...): SourceDefinition[] {
  return inputs.map(source => {
    // Logic for the indivual entry in the map
  }
}
```
Another argument for not having the `BoxMap` as a parameter is, that the only thing that is done to the variable before calling this method, is initializing it to an empty array. No need for that not to happen in this function.

Also, `BoxMap` is an array, so maybe not call it Map?